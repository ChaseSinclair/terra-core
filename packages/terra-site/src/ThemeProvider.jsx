import React from 'react';
import PropTypes from 'prop-types';

const propTypes = {
  /**
   * The component(s) that will be wrapped by <ThemeProvider />
   */
  children: PropTypes.node.isRequired,
  /**
   * Name of class for specified theme. e.g `terra-OSC-theme`
   */
  themeName: PropTypes.string.isRequired,
};

const ThemeProvider = ({ themeName, children, ...customProps }) => {
  return <div className={themeName}>{children}</div>;
};

ThemeProvider.propTypes = propTypes;

export default ThemeProvider;
