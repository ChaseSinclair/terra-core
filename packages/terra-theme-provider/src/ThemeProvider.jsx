import React from 'react';
import PropTypes from 'prop-types';
import CSSVariableProvider from './CSSVariableProvider';
import CSSVariablePolyfillProvider from './CSSVariablePolyfillProvider';

const propTypes = {
  /**
   * The component(s) that will be wrapped by <ThemeProvider />
   */
  children: PropTypes.node.isRequired,
  /**
   * Callback function that should return an array of strings to
   * themable stylesheet urls. e.g. `['https://cerner.com/styles-1032.css', 'https://cerner.com/styles-7A85.css']`
   */
  getThemeableCSS: PropTypes.func.isRequired,
  /**
   * Used to set new theme variables. e.g `{ '--terra-Badge-backgroundColor--default': '#bcbfc0' }`
   */
  theme: PropTypes.object,
};

const defaultProps = {
  theme: {},
};

const ThemeProvider = ({ theme, children, getThemeableCSS, ...customProps }) => {
  function supportsCSSVars() {
    return window.CSS && window.CSS.supports && window.CSS.supports('--fake-var', 0);
  }

  if (supportsCSSVars()) {
    return <CSSVariableProvider {...customProps} theme={theme}>{children}</CSSVariableProvider>;
  }

  return <CSSVariablePolyfillProvider {...customProps} theme={theme} getThemeableCSS={getThemeableCSS}>{children}</CSSVariablePolyfillProvider>;
};

ThemeProvider.propTypes = propTypes;
ThemeProvider.defaultProps = defaultProps;

export default ThemeProvider;
