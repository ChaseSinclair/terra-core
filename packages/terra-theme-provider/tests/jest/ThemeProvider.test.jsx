import React from 'react';
import ThemeProvider from '../../src/ThemeProvider';

describe('ThemeProvider', () => {
  const theme = {
    '--terra-badge-background-color-default': '#373d3f',
    '--terra-badge-color-default': '#fff',
  };

  const defaultRender = (
    <ThemeProvider theme={theme} getThemeableCSS={() => { document.querySelector('link[href*=terra-core]'); }}>
      <p>Child content</p>
    </ThemeProvider>
  );

  // Snapshot Tests
  it('should shallow render a theme provider component', () => {
    const wrapper = shallow(defaultRender);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render a theme provider component', () => {
    const wrapper = render(defaultRender);
    expect(wrapper).toMatchSnapshot();
  });

  // Error Handling Test
  it('should throw error for required children', () => {
    try {
      shallow(<ThemeProvider />);
    } catch (e) {
      expect(e.message).toContain('The prop `children` is marked as required');
    }
  });

  it('should throw error for required getThemeableCSS', () => {
    try {
      shallow(<ThemeProvider>test</ThemeProvider>);
    } catch (e) {
      expect(e.message).toContain('The prop `getThemeableCSS` is marked as required');
    }
  });
});
