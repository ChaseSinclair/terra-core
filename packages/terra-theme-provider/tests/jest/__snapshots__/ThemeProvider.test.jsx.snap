exports[`ThemeProvider should render a theme provider component 1`] = `
<div>
  <p>
    Child content
  </p>
</div>
`;

exports[`ThemeProvider should shallow render a theme provider component 1`] = `
<CSSVariablePolyfillProvider
  getThemeableCSS={[Function]}
  theme={
    Object {
      "--terra-badge-background-color-default": "#373d3f",
      "--terra-badge-color-default": "#fff",
    }
  }>
  <p>
    Child content
  </p>
</CSSVariablePolyfillProvider>
`;
